packages/common/src/content/manageSchedule.ts

```diff
@@ -0,0 +1,155 @@
+import { IRequestOptions } from "@esri/arcgis-rest-request";
+import { getHubApiUrl } from "../api";
+import { IHubSchedule, IHubScheduleResponse } from "../core/types/IHubSchedule";
+import { cloneObject } from "../util";
+import { deepEqual } from "../objects/deepEqual";
+import { IHubEditableContent } from "../core";
+
+// Any code referencing these functions must first pass isDownloadSchedulingAvailable
+
+/**
+ * Get the schedule for an item. If no schedule is found, returns null.
+ * @param itemId The item to get the schedule for
+ * @param requestOptions The request options needed to get the HubApiUrl
+ * @returns The schedule for the item OR null if no schedule is set
+ */
+export const getSchedule = async (
+  itemId: string,
+  requestOptions: IRequestOptions
+): Promise<IHubScheduleResponse> => {
+  const fetchResponse = await fetch(getSchedulerApiUrl(itemId, requestOptions));
+  const schedule = await fetchResponse.json();
+  if (!fetchResponse.ok || schedule.statusCode === 404) {
+    return {
+      message: `Download schedule not found for item ${itemId}`,
+      statusCode: 404,
+      schedule: null,
+    } as IHubScheduleResponse;
+  }
+
+  // if the schedule is set, return it with added mode
+  delete schedule.itemId;
+  switch (schedule.cadence) {
+    // TODO: add manual option here when option is viable
+    case "daily":
+    case "weekly":
+    case "monthly":
+    case "yearly":
+      return {
+        schedule: {
+          ...schedule,
+          mode: "scheduled",
+        },
+        message: `Download schedule found for item ${itemId}`,
+        statusCode: 200,
+      } as IHubScheduleResponse;
+  }
+};
+
+/**
+ * Set the schedule for an item
+ * @param itemId The item to set the schedule for
+ * @param schedule The schedule to set
+ * @param requestOptions The request options needed to get the HubApiUrl
+ */
+export const setSchedule = async (
+  itemId: string,
+  schedule: IHubSchedule,
+  requestOptions: IRequestOptions
+): Promise<IHubScheduleResponse> => {
+  const body = cloneObject(schedule);
+  delete body.mode;
+  const url = getSchedulerApiUrl(itemId, requestOptions);
+  const options = {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      accept: "application/json",
+    },
+    body: JSON.stringify({
+      ...body,
+      itemId,
+    }),
+  };
+  const response = await fetch(url, options);
+  return (await response.json()) as IHubScheduleResponse;
```

Comment:
Won't the API return `itemId` in the response? And won't we need to add `mode` like you do in `getSchedule`?

Reply:
Yes, I would do just that. Remember, these functions are supposed to return an `IHubScheduleResponse`. Our duty as developers is to make sure that callers of this function get a reponse object _always matches the return type_. Right now we're breaking the contract.